vent/io/hal.py

```diff
@@ -107,23 +110,48 @@ def secondary_pressure(self) -> float:
             raise RuntimeWarning('Secondary pressure sensor not instantiated. Check your "devices.ini" file.')
 
     @property
-    def flow(self) -> float:
-        """ The measured flow rate.
+    def flow_in(self) -> float:
+        """ The measured flow rate inspiratory side.
+        """
+        self._flow_sensor_in.update()
+        return self._flow_sensor_in.get()
+
+    @property
+    def flow_ex(self) -> float:
+        """ The measured flow rate expiratory side.
+        """
+        self._flow_sensor_en.update()
+        return self._flow_sensor_en.get()
+
+    @property
+    def setpoint_ex(self) -> float:
+        """ The currently requested flow on the expiratory side. This is solenoid open/close.
+
+        Returns:
+            float: 0<=setpoint<=1; The current set-point for flow control as a proportion of the maximum.
+        """
+        pass
+
+    @setpoint_ex.setter
+    def setpoint_ex(self, value: float):
+        """
+
+        Args:
+            value: Requested flow, as a proportion of maximum. Must be in [0, 1].

```

Comment:
Same here. This can either set the expiratory valve to open for `value < 0.5` and closed for` 0.5 <= value`  OR I can restrict `value` to take only 0 or 1.